engines/wasmer/wasmer.go

```diff
@@ -95,10 +100,22 @@ type (
 var _ = (wapc.Module)((*Module)(nil))
 var _ = (wapc.Instance)((*Instance)(nil))
 
-var engineInstance = engine{}
+type EngineOption func(e *engine)
 
-func Engine() wapc.Engine {
-	return &engineInstance
+func Engine(opts ...EngineOption) wapc.Engine {
+	e := engine{}
+	for _, opt := range opts {
+		opt(&e)
+	}
+	return &e
+}
+
+func WithMetrics(maxInstructions uint64, pfn unsafe.Pointer) EngineOption {
```

Comment:
We can probably be more specific, namely this is mostly about metering, right?
https://github.com/wasmerio/wasmer/blob/ea12ec0a0646325500588b42c48ac91a252649ae/lib/c-api/src/wasm_c_api/unstable/middlewares/metering.rs#L176-L179

```suggestion
func WithPushMeteringMiddlewarePtr(initialLimit uint64, pfn unsafe.Pointer) EngineOption {
```

PS this is one reason why I think it might be better to instead expose `WithConfig` as then you don't need to worry about the syntax and whether any typos matching config vals. Plus signature changes later don't end up rev-locking as most folks config types themselves don't move even if the values in them do.

Reply:
I would consider WithConfig better than WithPushMeteringMiddlewarePtr as the later is more an implementation detail. As for the limit, I am not certain how termination occurs and will try to find out exactly.  My main goal for this is to enable the metering which will allow our fabric to bill tenants on usage.  Termination for us will occur is there is insufficient blockchain gas regardless. So the termination is not my concern but I will try to find it for others sake.

Reply:
thanks for the clarification and the use case description! (ps I think what mainly got me into rename mode was metrics -> metering)